app/controllers/workouts_controller.rb

```diff
@@ -1,30 +1,16 @@
 class WorkoutsController < ApplicationController
-  before_action :set_workout, only: [:show, :update]
-
   def show
+    @workout = Workout.find(params[:id])
     @latest_workouts = Workout.order(created_at: :desc).first(7)
   end
 
   def index
-    Workout.all.find { |workout| @workout = workout if workout.todays_workout? }
-    @workout.nil? ? create : (redirect_to workout_path(@workout))
-  end
+    @workout = Workout.find_by(created_at: Time.zone.now.beginning_of_day..Time.zone.now.end_of_day)
 
-  def create
-    @workout = Workout.create
-    @workout.add_exercises
-    redirect_to workout_path(@workout)
-  end
+    CreateWorkoutService.create if @workout.nil?
 
-  def update
-    @workout.exercises.clear
-    @workout.add_exercises
-    redirect_to workout_path(@workout)
-  end
-
-  private
+    RemakeWorkoutService.new(@workout).remake if params[:remake]
 
-  def set_workout
-    @workout = Workout.find(params[:id])
+    redirect_to workout_path(Workout.last)
```

Comment:
You should never rely on an order to get the data. Always rely on the object directly or get it from the id